application-analytics-default/src/main/java/com/xwiki/analytics/script/AnalyticsScriptService.java

```diff
@@ -37,36 +40,48 @@
  * Script service for the Analytics Application.
  *
  * @version $Id$
- * @since 1.1
+ * @since 1.0
  */
 @Component
 @Named("analytics")
 @Unstable
 @Singleton
 public class AnalyticsScriptService implements ScriptService
 {
+    private static final String INVALID_MANAGER = "The manager couldn't be retrieved! Check if the hint is valid!";
+
     @Inject
-    @Named("Matomo")
-    private AnalyticsManager analyticsManager;
+    private Logger logger;
 
     @Inject
     private AnalyticsConfiguration configuration;
 
+    @Inject
+    private Provider<List<AnalyticsManager>> managerProvider;
     /**
      * Get data from Matomo API, in a format specific to the macro that will use it.
      *
-     * @param jsonNormaliserHint hint specific to the component that will normalize the Matomo result response,
-     *     since it's given / resulted format depends on the context were is used.
-     * @param filters holds the criteria for filtering a dataset.
      * @param parameters a map of the parameters needed for this request
-     * @return response from Matomo API, in a normalized JSON format
+     * @param filters holds the criteria for filtering a dataset.
+     * @param jsonNormaliserHint hint specific to the component that will normalize the response, since it's given /
+     * resulted format depends on the context were is used.
+     * @param managerHint hint specific to the service that will request the data
+     * @return a normalized JSON format
      */
-    @Unstable
-    public JsonNode getMatomoRequestResult(Map<String, String> parameters, Map<String, String> filters,
-        String jsonNormaliserHint)
+    public JsonNode makeRequest(Map<String, String> parameters, Map<String, String> filters,
+        String jsonNormaliserHint, String managerHint)
```

Comment:
I don't think we need the manager hint at this point because we don't plan to have a different implementation any time soon. It's fine to inject and use directly the Matomo implementation.